internal/workflow/activity/update_watermark.go

```diff
@@ -30,8 +31,9 @@ type (
 	}
 
 	UpdateWatermarkRequest struct {
-		Tag         uint32
-		BlockHeight uint64
+		Tag           uint32
+		BlockHeight   uint64
+		ValidateSince uint64
```

Comment:
nit: maybe validateStart would be a better name.  Please also add a comment on whether this is inclusive or exclusive index.